File: backend/utils/logger.js
/**
 * Logger utility - only logs in development mode
 * In production, these become no-ops
 */

const isDevelopment = process.env.NODE_ENV === 'development';

export const logger = {
    log: (...args) => {
        if (isDevelopment) {
            console.log(...args);
        }
    },

    error: (...args) => {
        // Always log errors, even in production
        console.error(...args);
    },

    warn: (...args) => {
        if (isDevelopment) {
            console.warn(...args);
        }
    },

    info: (...args) => {
        if (isDevelopment) {
            console.info(...args);
        }
    }
};

File: frontend/lib/utils/logger.dart
import 'package:flutter/foundation.dart';

/// Logger utility - only prints in debug mode
/// In release builds, these become no-ops
class AppLogger {
  static void log(String message) {
    if (kDebugMode) {
      debugPrint(message);
    }
  }

  static void error(String message, [dynamic error, StackTrace? stackTrace]) {
    // Always log errors
    debugPrint('❌ ERROR: $message');
    if (error != null) debugPrint('Details: $error');
    if (stackTrace != null) debugPrint('Stack: $stackTrace');
  }

  static void info(String message) {
    if (kDebugMode) {
      debugPrint('ℹ️ $message');
    }
  }

  static void success(String message) {
    if (kDebugMode) {
      debugPrint('✅ $message');
    }
  }

  static void warning(String message) {
    if (kDebugMode) {
      debugPrint('⚠️ $message');
    }
  }
}
